pkg/network/common/egress_dns.go

```diff
@@ -117,43 +121,34 @@ func (e *EgressDNS) Sync() {
 				// Item needs to wait for this duration before it can be processed
 				duration = tm.Sub(now)
 			} else {
-				changed, err := e.Update(dnsName)
-				if err != nil {
-					utilruntime.HandleError(err)
-				}
-
-				if changed {
-					e.Updates <- updates
-				}
-				continue
+				go e.Update(dnsName)
```

Comment:
why not just run go routine for doing the dns query?

Reply:
Prior to this PR we have the function `egressDNS.Sync()`, which calls `egressDNS.Update()`, which calls `DNS.Update()`. which gets blocked on `DNS.getMinIPsAndMinTTL()`.

Perhaps I'm missing something; but in order to do one goroutine per query the only two functions that can be ran in a separate goroutine are `egressDNS.Update()` or `DNS.Update()` because if we call `DNS.getMinIPsAndMinTTL()` in a separate goroutine `DNS.Update()` will still need to wait for it prior to doing the return. So instead of blocking in the function we'll block on waiting for it to write to the channel.

We could call `DNS.Update()` instead but then it would need to write to the `EgressDNS.dnsResponse` channel directly and I don't see any actual benefit. It would be harder to test and we would need to modify the tests which I managed to get without modifications.